ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -262,6 +266,46 @@ private User findByFilter(final BaseFilter filter, final boolean includeDeactiva
         }
     }
 
+    @Beta
+    @Nonnull
+    @Override
+    public List<User> findAllByQuery(@Nonnull final SearchQuery query) throws DaoException {
+        // build filter from search query
+        final List<BaseFilter> filters = new ArrayList<BaseFilter>();
```

Comment:
I don't, I prefer using the simplified generic form in Java 7 `new ArrayList<>()`, but this project currently needs support for Java 6